app/src/main/java/com/x8bit/bitwarden/data/auth/repository/AuthRepositoryImpl.kt

```diff
@@ -1256,6 +1259,33 @@ class AuthRepositoryImpl(
                 },
             )
 
+    override suspend fun validateEmailToken(email: String, token: String): EmailTokenResult {
+        return identityService
+            .verifyEmailRegistrationToken(
+                body = VerifyEmailTokenRequestJson(
+                    email = email,
+                    token = token,
+                ),
+            ).fold(
+                onSuccess = {
+                    when (val json = it) {
+                        is VerifyEmailTokenResponseJson.Invalid -> {
+                            if (json.message.contains(other = "expired", ignoreCase = true)) {
```

Comment:
Could this check be done in the service, and the `VerifyEmailTokenResponseJson` could have a 3rd type of `Expired`.

Reply:
> I mainly want to hide the check for a specific string.

Pre-edit, I was indifferent, but the rationale of hiding the "ugliness" I am 110% sold.